src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1392,7 +1392,7 @@ static bool CopyFile_CopyFileRange(int inFd, int outFd, const struct stat_ *sour
     {
         ssize_t copied = syscall(PAL_COPY_FILE_RANGE_SYSCALL, inFd, NULL, outFd, NULL, (size_t)size, 0);
 
-        if (size < 0)
+        if (copied < 0)
```

Comment:
Ooh, good catch!

Reply:
There were some looping tests on my first attempt. This seemed to be the cause.